trulens_eval/trulens_eval/feedback/groundtruth.py

```diff
@@ -335,3 +341,277 @@ def rouge(
             ret = np.nan
 
         return ret
+
+
+class GroundTruthAggregator(WithClassInfo, SerialModel):
+    model_config: ClassVar[dict] = dict(
+        arbitrary_types_allowed=True, extra="allow"
+    )
+    """Aggregate benchmarking metrics for ground-truth-based evaluation on feedback fuctions."""
+
+    true_labels: List[int]
+    custom_agg_funcs: Dict[str, Callable] = pydantic.Field(default_factory=dict)
+
+    k: Optional[int] = (
+        None  # top k results to consider in NDCG@k, precision@k, recall@k, etc
+    )
+
+    n_bins: int = 5  # number of bins for ECE
+
+    def __init__(
+        self, true_labels: List[int], k: Optional[int] = None, **kwargs
+    ):
+        # TODO: automatically load from IR / benchmarking datasets or just set the DB url for smaller serialization overhead?
+        super().__init__(
+            true_labels=true_labels, k=k, custom_agg_funcs={}, **kwargs
+        )
+
+    def register_custom_agg_func(
```

Comment:
@sfc-gh-pdharmana  I added a feature to allow user to register custom aggregation functions. 

The implications on serialization should be fine as I think this is mainly going to be only used on the client side. 
 I haven't given this too much deep considerations tho, but one thing I can take care in the next PR is more scalable treatment than loading all GT labels (`self.true_labels`) into the memory.

Reply:
Can you add an example for this?

Reply:
yes it's already in the `meta_evaluation_examples` notebook (last 2 cells)